arangod/Aql/Executor/DistributeExecutor.cpp

```diff
@@ -121,8 +124,14 @@ auto DistributeExecutorInfos::shouldDistributeToAll(
   return false;
 }
 
+ResourceMonitor& DistributeExecutorInfos::resourceMonitor() const noexcept {
+  return _resourceMonitor;
+}
+
 DistributeExecutor::DistributeExecutor(DistributeExecutorInfos const& infos)
-    : _infos(infos) {}
+    : _infos(infos),
+      _temp(std::make_shared<velocypack::SupervisedBuffer>(
```

Comment:
Can't the supervised buffer be built on the stack as _temp is a class member, is gonna linger there, and no one's gonna steal the buffer from it?